core/app/[locale]/(default)/account/(tabs)/order/[slug]/page-data.tsx

```diff
@@ -0,0 +1,199 @@
+import { graphql, ResultOf, VariablesOf } from '~/client/graphql';
+import { revalidate } from '~/client/revalidate-target';
+import { ExistingResultType } from '~/client/util';
+
+import {
+  OrderItemFragment,
+  ProductAttributes,
+  ProductAttributesVariables,
+} from '../../orders/_components/product-snippet';
+
+export const OrderShipmentFragment = graphql(`
+  fragment OrderShipmentFragment on OrderShipment {
+    shippingMethodName
+    shippingProviderName
+    tracking {
+      __typename
+      ... on OrderShipmentNumberAndUrlTracking {
+        number
+        url
+      }
+      ... on OrderShipmentUrlOnlyTracking {
+        url
+      }
+      ... on OrderShipmentNumberOnlyTracking {
+        number
+      }
+    }
+  }
+`);
+
+const CustomerOrderDetails = graphql(
+  `
+    query CustomerOrderDetails($filter: OrderFilterInput) {
+      site {
+        order(filter: $filter) {
+          entityId
+          orderedAt {
+            utc
+          }
+          status {
+            label
+            value
+          }
+          totalIncTax {
+            value
+            currencyCode
+          }
+          subTotal {
+            value
+            currencyCode
+          }
+          discounts {
+            nonCouponDiscountTotal {
+              value
+              currencyCode
+            }
+            couponDiscounts {
+              couponCode
+              discountedAmount {
+                value
+                currencyCode
+              }
+            }
+          }
+          shippingCostTotal {
+            value
+            currencyCode
+          }
+          taxTotal {
+            value
+            currencyCode
+          }
+          billingAddress {
+            firstName
+            lastName
+            address1
+            city
+            stateOrProvince
+            postalCode
+            country
+          }
+          consignments {
+            shipping {
+              edges {
+                node {
+                  entityId
+                  shippingAddress {
+                    firstName
+                    lastName
+                    address1
+                    city
+                    stateOrProvince
+                    postalCode
+                    country
+                  }
+                  shipments {
+                    edges {
+                      node {
+                        entityId
+                        shippedAt {
+                          utc
+                        }
+                        ...OrderShipmentFragment
+                      }
+                    }
+                  }
+                  lineItems {
+                    edges {
+                      node {
+                        ...OrderItemFragment
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `,
+  [OrderItemFragment, OrderShipmentFragment],
```

Comment:
I feel like order details should be a framgent coming from `order-details.tsx`.

Reply:
My logic was that `<OrdersDetails/>` component is basically aggregates smaller components like:
  - `<OrderState/>`
  - `<ProductSnippet/>`  => which presents Order's consignment(s)
  - `<OrderSummaryInfo/>`
 - ` <ShippingInfo/>`
  - `<PaymentInfo/>`  => will be added later.


  So basically we fetch `CustomerOrderDetails` data and map it in suitable for `<OrdersDetails/>` format on `page.tsx`.